### DISCUSSION TITLE
Security flaw: Stump takes orders from X by default

### DISCUSSION THREAD

**Discussion by uti5** [stumpwm/stumpwm#1121] [General]
One generally assumes that access to an X server does not entail arbitrary code
execution on behalf of its owner, yet this becomes the default with Stump.  It
would be more secure to require the user to enable commands through X
explicitly (after having seen a warning about the security implication).

**Comment by szos** [comment]
While I think having the ability to forbid arbitrary execution is a good move, I'm not sure what to think about disabling it by default. Theres a fair number of folks out there (myself included) who use the X11 property to communicate with stump (either through stumpish or through some other method like xprop), and i think its generally a bad move to break peoples configs. I think the most unobtrusive method would be to have a variable such that the user can control whether or not the X11 property gets read for commands to execute, but this also means anyone who doesnt read the documentation is going to leave this enabled. 

One idea could be to have a variable controlling this with three states, `nil` `:disable` and `:enable`, with it being `nil` by default. When the `:property-notify` event handler encounters a `:stumpwm_command` or `:rp_command_request` atom, if the variable is `:disable` it just ignores the request, if its `:enable` function as per usual, and if its `nil` message the user that command evaluation via X11 is disabled by default and they need to set the variable to `:enable`. This would keep users who dont bother reading the docs informed, and while it would be a hiccup for them, it would only require adding one line to their stumpwmrc. 

thoughts?

**Comment by adlai** [comment]
I realise that repeating the quote, "you're using Common Lisp, why are you worried about security?" is pretty much the opposite of helpful, although it is an interesting perspective.

In this case, if your hypethetical attacker already has control of the X server, wouldn't it be almost trivial to select any scripting language installed on the computer? Blaming the stump process for being the one common lisp image that happened to get chosen is slightly myopic, in my opinion.

**Comment by dmb2** [comment]
I will entertain PRs but I don't want to keep this open.

**Comment by uti5** [comment]
@adlai, how is it ``trivial to select ...''?  What is the point of `xhost', if access to an X server is (supposed to be) complete control of the owner?  With stump defaults, this doesn't make sense:

https://wiki.gentoo.org/wiki/Simple_sandbox

**Comment by adlai** [comment]
I'll unsubscribe and keep reading after I'm not quite so ignorant.
